pkg/controller/nodeipam/ipam/controller.go

```diff
@@ -178,6 +178,7 @@ func (c *Controller) onAdd(node *v1.Node) error {
 		syncer = c.newSyncer(node.Name)
 		c.syncers[node.Name] = syncer
 		go syncer.Loop(nil)
+		syncer.Update(node) // new node, trigger sync on add
```

Comment:
This is common to both if/else - you can move it out.

Reply:
yeah, was thinking about this. The change would be to drop the else and do Update in all cases, will also involve changing the warning label accordingly. Will make the change.